app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,25 @@
+class SessionsController < ApplicationController
+  before_action :find_user, only: :create
+  def new; end
+
+  def create
+    if @user&.authenticate params[:session][:password]
+      log_in @user
+      params[:session][:remember_me] == "1" ? remember(@user) : forget(@user)
+      redirect_to @user
+    else
+      flash.now[:danger] = t "invalid"
+      render :new
+    end
+  end
+
+  def destroy
+    log_out if logged_in?
+    redirect_to root_url
+  end
+
+  private
+  def find_user
+    @user = User.find_by email: params[:session][:email].downcase
```

Comment:
xử lý nếu user bị nil tại đây luôn cu 
```suggestion
    @user = User.find_by email: params.dig(:session, :email)&.downcase
```